### PR TITLE
Apex-1631 - szulinap bekeres ujrapingeles

### PR DESCRIPTION
**Pull request by attilazsoltdanku**
The birthday bot will ask again the birthdays if they are missing.
https://apexlab.atlassian.net/jira/software/c/projects/APEX/boards/25?assignee=62f0f4241323922c61e35810&issueParent=17482&selectedIssue=APEX-1631

### CODE REVIEW

File: packages/functions/events/askBirthday.ts [hunk 1]
```diff
@@ -35,6 +36,12 @@ export const handler = handleEvent(
         return;
       }
 
+      await saveBirthday({
+        birthday: null,
+        teamId: userInfo.team_id || "",
```
Comment (BaDo2001) [inline packages/functions/events/askBirthday.ts]: I would also include this in the check above and not save if team_id is not present. (Even though I don't think its very likely)

**Review by BaDo2001 — CHANGES_REQUESTED** [review]
Nice job with the feature!!

However I don't see any tests related to the change.

In the askBirthday integration tests we should validate that the blank user is created when the message is sent out.

We should also write new integration tests that validate that this message is sent out when the daily cron is ran. (I have some code which can be used for that so lets talk)

File: tests/integration/askBirthday.test.ts [hunk 2]
```diff
@@ -207,4 +207,29 @@ describe("Slack interactions", () => {
...
+
+      await testDb.insert(users).values([
+        {
+          id: import.meta.env.VITE_SLACK_USER_ID,
+          teamId: import.meta.env.VITE_SLACK_TEAM_ID,
+          birthday: null,
+        },
+      ]);
+
+      await fetch(`${import.meta.env.VITE_API_URL}/daily?eventId=${eventId}`);
```
Comment (BaDo2001) [inline tests/integration/askBirthday.test.ts]: Please update this to use the publishCronEvent util.

File: tests/unit/askBirthday.test.ts [hunk 3]
```diff
@@ -81,3 +101,35 @@ describe("Member joined channel", () => {
...
+    const result = await getUsersWhoseBirthdayIsMissing();
+
+    expect(result).toEqual(mockUser);
+  });
+
+  it("shouldn't find any users", async () => {
+    await testDb
+      .update(users)
+      .set({ birthday: dayjs.utc().toDate() })
+      .where(eq(users.id, mockUser[0].id));
```
Comment (BaDo2001) [inline tests/unit/askBirthday.test.ts]: This test is fake.

You are clearing the db in the aftereach, so this update is not going to find a user, so the result will always be empty.

In general tests should be independent, they should setup separately (i.e create user in db)

**Review by BaDo2001 — CHANGES_REQUESTED** [review]

File: tests/integration/askBirthday.test.ts (hunk 2 above)
Comment (attilazsoltdanku) [reply tests/integration/askBirthday.test.ts]: Nice catch :D

File: tests/unit/askBirthday.test.ts (hunk 3 above)
Comment (attilazsoltdanku) [reply tests/unit/askBirthday.test.ts]: Nice catch :D

**Review by BaDo2001 — APPROVED** [review]

**Pull request merged**